catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/Gml3ToWktImpl.java

```diff
@@ -59,26 +71,60 @@ public String convert(String xml) throws ValidationExceptionImpl {
     }
   }
 
-  @SuppressWarnings("unchecked")
-  public String convert(InputStream xml) throws ValidationExceptionImpl {
-    AbstractGeometryType geometry = null;
+  public String convert(InputStream xml) throws ValidationException {
+    return convert(xml, null);
+  }
+
+  public String convert(InputStream xml, Function<Object, String> unknownClassCallback)
+      throws ValidationException {
+    if (latLonTransform == null) {
+      LOGGER.debug("Lat/Lon transform is null");
+      throw new ValidationExceptionImpl("Unable to transform geometry due to null transform");
+    }
+
+    Object parsedObject = parseXml(xml);
+
+    if (parsedObject instanceof Envelope) {
+      parsedObject = JTS.toGeometry((Envelope) parsedObject);
+    }
+
+    if (parsedObject instanceof Geometry) {
+      try {
+        Geometry geometry = JTS.transform((Geometry) parsedObject, latLonTransform);
+        return wktWriter.write(geometry);
+      } catch (TransformException e) {
+        LOGGER.debug("Failed to transform geometry to lon/lat", e);
+        throw new ValidationExceptionImpl(
+            e,
+            Collections.singletonList("Cannot transform geometry to lon/lat"),
+            new ArrayList<>());
+      }
+    } else if (unknownClassCallback != null) {
```

Comment:
Using a callback to do the processing this class is supposed to do feels a little akward to me. Wondering if instead this bean had a service list for an interface of GmlHandlers and if it's not a geometry class we go through our list of GmlHandler services until one can handle the object.